### PR TITLE
Add delimiter parameter to readDelimStr

### PR DESCRIPTION
**Pull request by koperagen**

### CODE REVIEW

File: tests/src/test/kotlin/org/jetbrains/kotlinx/dataframe/samples/api/Write.kt [hunk 1]
```diff
@@ -46,7 +46,8 @@ class Write : TestBase() {
     @Test
     fun writeCsvStr() {
         // SampleStart
-        val csvStr = df.toCsv(CSVFormat.DEFAULT.withDelimiter(';').withRecordSeparator(System.lineSeparator()))
+        val format = CSVFormat.DEFAULT.builder().setDelimiter(';').setRecordSeparator(System.lineSeparator()).build()
```
Comment (Jolanrensen) [inline tests/src/test/kotlin/org/jetbrains/kotlinx/dataframe/samples/api/Write.kt:49]: long line, but the linter will fix all these soon anyway

**Review by Jolanrensen — APPROVED** [review]
Yes! I missed that parameter too. Looks like TC still fails, but after the test is fixed, should be good to go

**Comment by dave08** [comment]
Little comment, maybe since this is a function to especially read delimeters, it might be useful to have an override that takes in a Regex as a delimiter... this might be used for command line output tables that are usually space separated but sometimes inside a column value there might be a single space, so I need to use "\\s\\s+" to correctly read it in.

**Comment by Jolanrensen** [comment]
@dave08 sounds like a good idea :) could you make an issue for that?

**Comment by dave08** [comment]
https://github.com/Kotlin/dataframe/issues/746

**Pull request merged**